tests/test_auth_lock.py

```
@@ -0,0 +1,83 @@
+    if lock_file.exists():
+        os.remove(lock_file)
+
+    with patch("run_bot.config.auth_lock_path", str(lock_file)):
+        with patch("sys.argv", ["run_bot.py", "--max-days", "1"]):
+            # For non-credential error, run_bot might continue or exit depending on other logic,
+            # but it should NOT exit with code 2 and NOT create a lock file.
+            try:
+                run_bot.main()
+            except SystemExit as e:
+                assert e.code != 2
+            except Exception:
```

Pylint (4.0.6)

[warning] Catching too general exception Exception

(W0718)